tests/test_cloud_init_iso.cpp

```diff
@@ -421,13 +423,77 @@ TEST_F(CloudInitIso, updateCloudInitWithNewEmptyExtraInterfaces)
                                                                                       iso_path.toStdString()));
     mp::CloudInitIso new_iso;
     new_iso.read_from(iso_path.toStdString());
-    EXPECT_FALSE(new_iso.contains("network-config"));
+    EXPECT_TRUE(new_iso.contains("network-config"));
+}
+
+TEST_F(CloudInitIso, updateCloneCloudInitSrcFileWithExtraInterfaces)
+{
+    constexpr auto* src_meta_data_content = R"(#cloud-config
+instance-id: vm1_e_e
+local-hostname: vm1
+cloud-name: multipass)";
+    constexpr auto* src_network_config_data_content = R"(#cloud-config
+version: 2
+ethernets:
+  default:
+    match:
+      macaddress: "00:00:00:00:00:00"
+    dhcp4: true
+    dhcp-identifier: mac
+  extra0:
+    match:
+      macaddress: "00:00:00:00:00:01"
+    dhcp4: true
+    dhcp-identifier: mac
+    dhcp4-overrides:
+      route-metric: 200
+    optional: true
+)";
```

Comment:
Could the template be extracted here and below and then used with `fmt::format` to fill in MACs?

Reply:
Nice catch, it should be done in that.